src/app/ui/comment/comment.component.ts

```diff
@@ -21,11 +23,16 @@ export class CommentComponent implements OnInit {
   allowToMarkAsSolution: boolean;
 
   @Input()
-  topic: TopicId;
+  topic: TopicWithCreatorId;
 
-  constructor(private db: DbService, private snackBar: MatSnackBar) { }
+  canMark: boolean;
+  isAdmin: boolean;
+
+  constructor(private db: DbService, private snackBar: MatSnackBar, private auth: AuthService) { }
 
   ngOnInit() {
+    this.auth.isAdminForCourse(this.topic.linkedCourseId).subscribe(is => this.isAdmin = is);
```

Comment:
Wrong use of observables - please read again how to work with them. You calculate canMark once and before you have the result of the observable...